fw/sock_clnt.c

```diff
@@ -152,6 +152,18 @@ tfw_cli_conn_send(TfwCliConn *cli_conn, TfwMsg *msg)
 {
 	int r;
 
+	/*
+	 * msg can be freed in ss_do_send callback and
+	 * if it is the last msg for this connection
+	 * connection reference counter became equal to 1.
+	 * But if client close conenction and send FIN to us
+	 * connection reference counter will be decremented
+	 * and connection will be freed, so we should increment
+	 * connection reference counter here to access `cli_conn`
+	 * after sending data in this function.
+	 */
```

Comment:
Please allow me to rephrase a little bit. :-) You are free to use this any way you want.

_msg can be freed in ss_do_send() callback. If it's the last msg for this connection the ref counter becomes equal to 1. However if the client closes the connection and sends FIN, the connection ref counter is decremented to zero and the connection is freed. To avoid that, increment the connection ref counter here to guarantee access to cli_conn after sending data._

Reply:
Yes fixed.